File: Data-Structures-And-Algorithms/Others/leetCode-121-maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        int diff = 0;
        int low = prices[0];
        for (int i=1; i < prices.length; i++) {
            if (prices[i] < low) {
                low = prices[i];
            }
            if (prices[i] - low > diff) {
                diff = prices[i] - low;
            }
        }
        return diff;
    }
}

File: Data-Structures-And-Algorithms/Others/leetCode-121-maxProfit.go
func maxProfit(prices []int) int {
	low, diff := prices[0], 0
	for i := 1; i < len(prices); i++ {
		if prices[i] < low {
			low = prices[i]
		}
		if prices[i]-low > diff {
			diff = prices[i] - low
		}
	}
	return diff
}
